source/host_resolver.c

```diff
@@ -80,6 +82,12 @@ int aws_host_resolver_purge_cache(struct aws_host_resolver *resolver) {
     return resolver->vtable->purge_cache(resolver);
 }
 
+int aws_host_resolver_purge_host_cache(const struct aws_host_resolver_purge_host_options *options) {
+    AWS_ASSERT(options && options->resolver);
```

Comment:
This a external API, people can really pass NULL to this function or invalid options.

We should do a raise error instead for proper error handling.

Reply:
Thanks, makes sense. I have updated it to throw invalid argument error.